ocrd_eval.schema.yml

```diff
@@ -0,0 +1,217 @@
+        description: The workspace containing the GT
+
+      workflow_steps:
+        type: array
+        description: Human readable description of the individual steps and their parameters in the workflow (for UI)
+        minItems: 1
+        items:
+          type: object
+          properties:
+            id:
+              type: string
+              description: The name of the processor used for this workflow step
+              pattern: '^ocrd-[a-z\-]+'
+            params:
+              type: object
+              description: A map of parameters and their values applied to the processor used for this workflow step
+          required: ['id', 'params']
+
+      workflow_model:
+        type: string
+        description: Human readable name of the main model used for recognition in the OCR workflow (for UI)
+
+      eval_tool:
+        type: string
+        description: Human readable name and version of evaluation tool used (for UI)
+
+      document_metadata:
+        type: object
+        title: Bibliographical and typographical metadata about the work to be evaluated
+        properties:
+
+          publication_year:
+            type: number
+            description: Year the document was originally published
+
+          publication_century:
+            type: string
+            description: Century the document was originally published
+            pattern: '[12][0-9]{3}-[12][0-9]{3}'
+
+          publication_decade:
+            type: string
+            description: Decade the document was originally published
+            pattern: '[12][0-9]{2}0-[12][0-9]{2}0'
+
+          number_of_pages:
+            type: number
+            description: Number of pages in this work (i.e. the number of images in the gt_workspace)
+
+          layout:
+            type: string
+            enum: ['simple', 'complex']
+
+          fonts:
+            type: array
+            items:
+              type: string
+              enum: ['antiqua', 'fraktur']
+
+      provenance:
+        type: object
+        description: Information on which tools in which version were used in determining metrics
+        properties:
+          parameters:
+            type: object
+            description: Parameters passed to the evaluation processor
+
+  EvaluationReport:
+    type: object
+    additionalProperties: false
+    description: The metrics measured for this document
+    properties:
+      document_wide:
+        type: object
+        description: Document-wide metrics
+        allOf: [
+          { $ref: '#$defs/DocumentEvaluationMetrics' },
+          { $ref: '#$defs/CommonEvaluationMetrics' }
+        ]
+        unevaluatedProperties: false
+      by_page:
+        type: array
+        description: Metrics page-by-page
+        items:
+          type: object
+          allOf: [
+            { $ref: '#$defs/CommonEvaluationMetrics' },
+            { $ref: '#$defs/PageId' }
+          ] 
+          unevaluatedProperties: false
```

Comment:
This might not be the best solution, but it's the only one I managed so far to achieve a strict validation of the properties.
Open to suggestions!

Reply:
> > `unevaluatedProperties` is like `additionalProperties`, except that it can "see through" `$ref` and "see inside" `allOf`, `anyOf`, `oneOf`, `if`, `then`, `else`, and whatever else I'm forgetting right now.

(https://github.com/json-schema-org/json-schema-spec/issues/556)

I use `additionalProperties: false` all the time and IIUC `unevaluatedProperties` offers the same but can traverse referenced subschemas, so I think this is actually is the best solution.

Reply:
thanks for the feedback!